### DISCUSSION TITLE
MIME type offered for webp downloads is possibly incorrect?

### DISCUSSION THREAD

**Discussion by lbeltrame** [vladmandic/sdnext#2750] [Q&A] [answered]
Posting here as I'm not sure if it's an issue in SD.Next or upstream Gradio.

I've noticed that if you set webp as image format, the MIME type of the download is wrongly set as `text/plain` instead of `image/webp`:

```
curl -s -I "http://127.0.0.1:7860/file=/tmp/gradio/2aa08251e8db3cbf16cd04f036175e65ac4c0952/00084-256793276-animagine-xl-3.0-1girl%20solo%20standing%20leaning%20forward%20arms%20behind%20back.webp" | grep -i "^Content-Type:"
content-type: text/plain; charset=utf-8
```

But at least looking at the file's magic number identifies it as webp:

```
file 00084-256793276-animagine-xl-3.0-1girl\ solo\ standing\ leaning\ forward\ arms\ behind\ back.webp 
00084-256793276-animagine-xl-3.0-1girl solo standing leaning forward arms behind back.webp: RIFF (little-endian) data, Web/P image
```

This is problematic for example if you want to download images from mobile, as they will be interpreted as text. However I'm not sure if this is a problem with Gradio (a quick glance at their issue tracker did not show anything) or in SD.Next itself.

**Comment by vladmandic** [comment, marked as answer]
it is a gradio thing, but i can still provide additional mime types from sdnext - its fixed in dev.
